201-vm-msi-linux-terraform/nestedTemplates/setUpRbac.json

```diff
@@ -0,0 +1,27 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+      "principalId": {
+        "type": "string",
+        "metadata": {
+          "description": "Principal ID to set the access for"
+        }
+      }
+    },
+    "variables": {
+      "owner": "[concat('/subscriptions/', subscription().subscriptionId, '/providers/Microsoft.Authorization/roleDefinitions/', '8e3af657-a8ff-443c-a75c-2fe8c4bcb635')]"
```

Comment:
any reason this can't be contributor instead of owner?

Reply:
I believe MSI requires owner. The other two samples which do MSI have the same permission level.

Reply:
MSI doesn't need owner perms, contrib will work as long as you're not doing anything that requires owner (can't imagine what that would be) - so let's use least privilege here.

Reply:
The owner was a variable name. Anyway I have changed the variable name  to contributor to avoid any confusion.